integration_tests/test_middlewares.py

```diff
@@ -0,0 +1,11 @@
+import pytest
+from utils import get
+
+
+@pytest.mark.skip(reason="Fix middleware request headers modification")
```

Comment:
This is broken?

Reply:
Yes, I started investigating it but I'm actually waiting for PR #375 to be merged because I think its new struct will help as we could make `Request` a pyclass and pass its instance to the middlewares (and routes) instead of a hashmap, and that should help fix this. We could also manipulate a true `Request` object in the routes and misslewares instead of a raw dict, which would be much nicer!